Move to PHPUnit 6

Slamdunk opened the pull request:
- [x] Migrate to PHPUnit 6 (thanks to [PHP-CS-Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer))
- [x] Drop `composer.lock`, so drop Travis `DEPS=locked`
- [x] Align dependencies to https://github.com/doctrine/DoctrineModule/pull/613
- [x] Test against PHP 7.2 and pass
- [x] Strip Travis duplicated builds
- [x] Make `CliController` output mockable

Supersedes https://github.com/doctrine/DoctrineModule/pull/610 and https://github.com/doctrine/DoctrineModule/pull/613

Majkl578 commented on tests/DoctrineModuleTest/Component/Console/Input/RequestInputTest.php:
```diff
@@ -18,7 +18,7 @@
 namespace DoctrineModuleTest\Component\Console\Input;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase as TestCase;
```
The alias is useless.

Majkl578 commented on tests/DoctrineModuleTest/Component/Console/Output/PropertyOutputTest.php:
```diff
@@ -18,7 +18,7 @@
 namespace DoctrineModuleTest\Component\Console\Output;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase as TestCase;
```
The alias is useless.

Majkl578 commented on tests/DoctrineModuleTest/Form/Element/ProxyAwareElementTestCase.php:
```diff
@@ -21,7 +21,7 @@
 use Doctrine\Common\Collections\ArrayCollection;
 use DoctrineModuleTest\Form\Element\TestAsset\FormObject;
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase as TestCase;
```
The alias is useless.

Majkl578 commented on tests/DoctrineModuleTest/Validator/Service/NoObjectExistsFactoryTest.php:
```diff
@@ -2,7 +2,7 @@
 namespace DoctrineModule\Validator\Service;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase as TestCase;
```
The alias is useless.

Majkl578 commented on tests/DoctrineModuleTest/Validator/Service/ObjectExistsFactoryTest.php:
```diff
@@ -2,7 +2,7 @@
 namespace DoctrineModule\Validator\Service;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase as TestCase;
```
The alias is useless.

Majkl578 commented on tests/DoctrineModuleTest/Validator/Service/UniqueObjectFactoryTest.php:
```diff
@@ -2,7 +2,7 @@
 namespace DoctrineModule\Validator\Service;
 
-use PHPUnit_Framework_TestCase as TestCase;
+use PHPUnit\Framework\TestCase as TestCase;
```
The alias is useless.

Slamdunk commented:
1. Useless aliases dropped
1. nightly build on latest deps reintroduced

> I don't really approve the changes you made to CI build matrix. If you want to refactor Travis build, please do so using Build Stages as we did in some repos (i. e DBAL, ORM) already - it's more readable and understandable.

I didn't refactored it, I just dropped duplicated builds, nightly and locked ones. Now nightly are reintroduced on latest.
I've done it so to ease the mergeability.

I am able to move travis to stages, but this repo seems a bit stalled, I don't want to create a huge PR that will take a lot of time to be understood and merged.

> I don't understand why you removed nightly, it's perfectly legit to test against php-next (7.3 now) to spot possible issues early. The fact it's failing now for master with PHP fatal error is not justification to remove it, it should be fixed if still relevant (maybe fixed by deps bump, else separately from this PR).

I've not dropped it because it failing: as you can see with my PR it passes without specific changes.

I've dropped it because at the time of writing there is no difference between 7.2 and nightly.
A new release is a year apart from now and we have no clue if the changes made in the next months will see the light or not. To me **at the time of writing** nightly build are just a waste of computational resources, and I'm trying to be as much environmentalist as I can.

Majkl578 commented:
> I've not dropped it because it was failing: as you can see with my PR it passes without specific changes.

Ok, coincidence - if you look at master, there are three failures (two different issues) on nightly: https://travis-ci.org/doctrine/DoctrineModule/builds/309067889

Looks like one failure was caused by old PHPUnit: https://travis-ci.org/doctrine/DoctrineModule/jobs/309067901
But not this one, with lowest deps on nightly: https://travis-ci.org/doctrine/DoctrineModule/jobs/309067899

> I am able to move travis to stages, but this repo seems a bit stalled, I don't want to create a huge PR that will take a lot of time to be understood and merged.

Fine, leave it as is. :)

TomHAnderson commented:
@Slamdunk this module is stalled.  I agree with your choice to move forward without stages and implement those once DoctrineModule is up to speed with PHP 7.1

Slamdunk commented:
> this module is stalled

Does this mean that the project will still be maintained with a PR such this or not?

I run a million euro business on DoctrineModule + DoctrineORMModule: I am intended to keep them functioning and up to date, help with PR like this.

Ocramius commented:
Yeah, it will be maintained, just not improved for now.

On 2 Jan 2018 09:16, "Filippo Tessarotto" <notifications@github.com> wrote:

> this module is stalled
>
> Does this mean that the project will still be maintained with a PR such
> this or not?
>
> I run a million euro business on DoctrineModule + DoctrineORMModule: I am
> intended to keep them functioning and up to date, help with PR like this.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/doctrine/DoctrineModule/pull/615#issuecomment-354718462>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAJakCqqpZ5RWgnvPjH93jZQ-HSOkkotks5tGeXDgaJpZM4ROaXl>
> .
>

TomHAnderson commented:
@Slamdunk I like this PR and I'll merge it into dev-master soon.  I don't have a timeline for a release at this time though.

Majkl578 commented:
Since PHPUnit 7 is out, would probably make sense to upgrade here too?

Slamdunk commented:
@Majkl578 Done :+1:

Slamdunk commented:
I've added 7.0 to composer, but zend-test disallow it

TomHAnderson commented:
ok, thanks for trying that @Slamdunk 

I'll dig into this PR tomorrow.

TomHAnderson commented:
Released with pre-release 2.0.1
